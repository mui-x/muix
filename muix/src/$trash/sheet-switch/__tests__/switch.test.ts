import { atomizeRuleset } from "reactxx-styler";
import { initPlatform, Shape, ts } from "reactxx-tests";

describe("SWITCH", () => {

  const doTest = (isWeb: boolean) => {
    beforeEach(() => initPlatform(isWeb))
    let ruleset

    it("01 isOpened: color: 'red', isClosed: color: 'blue'", () => {
      ruleset = atomizeRuleset<'Text', Shape>({
        $switch: {
          isOpened: {
            color: 'red'
          },
          isClosed: {
            color: 'blue'
          },
        }
      })
      expect(ruleset).toMatchSnapshot()
    })

    it("02 :hover => isOpened => :active => color: 'red'", () => {
      ruleset = atomizeRuleset<'Text', Shape>({
        $web: {
          ':hover': {
            $switch: {
              isOpened: {
                $web: {
                  ':active': {
                    color: 'red'
                  }
                }
              }
            }
          }
        }
      })
      expect(ruleset).toMatchSnapshot()
    })

    it("03 atomizeRuleset(isOpened => atomizeRuleset(color: 'red), isClosed => atomizeRuleset(color: 'green'))", () => {
      ruleset = atomizeRuleset<'Text', Shape>({
        $switch: {
          isOpened: atomizeRuleset({ color: 'red' }) as any,
          isClosed: atomizeRuleset({ color: 'green' }) as any,
        }
      })
      expect(ruleset).toMatchSnapshot()
    })

    it("04 variant's order: black-red-green-yellow-blue-brown", () => {
      ruleset = atomizeRuleset<'Text', Shape>({
        $web: {
          color: 'blue',
          $switch: {
            isOpened: {
              color: 'brown',
            }
          }
        },
        $switch: {
          isOpened: {
            color: 'red',
            $web: {
              color: 'yellow'
            },
            $switch: {
              isClosed: {
                color: 'green',
              }
            }
          },
        },
        color: 'black'
      })
      expect(ruleset).toMatchSnapshot()
    })

    it("05 variant's order with array: $web => brown-blue-yellow-red-black", () => {
      ruleset = atomizeRuleset<'Text', Shape>([{
        $web: [{
          $switch: {
            isOpened: {
              color: 'brown',
            }
          }
        }, {
          color: 'blue',
        }],
      }, {
        $switch: {
          isOpened: [{
            $web: {
              color: 'yellow'
            },
          }, {
            color: 'red',
          }],
        },

      }, {
        color: 'black'
      }])
      expect(ruleset).toMatchSnapshot()
    })

    it("06 nesting PSEUDO and $switch", () => {
      ruleset = atomizeRuleset<'Text', Shape>({
        $web: {
          ':hover': {
            $switch: {
              isOpened: {
                $web: {
                  ':active': {
                    margin: 10
                  }
                }
              },
            }
          }
        }
      })
      expect(ruleset).toMatchSnapshot()
    })

}

  describe("## NATIVE ##", () => doTest(false))
  describe("## WEB ##", () => doTest(true))
});
