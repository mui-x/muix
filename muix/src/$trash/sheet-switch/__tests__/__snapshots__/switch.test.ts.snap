// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SWITCH ## NATIVE ## 01 isOpened: color: 'red', isClosed: color: 'blue' 1`] = `
Array [
  Array [
    "IF $switch: isOpened",
    "color: red /*./$switch.isOpened*/",
  ],
  Array [
    "IF $switch: isClosed",
    "color: blue /*./$switch.isClosed*/",
  ],
]
`;

exports[`SWITCH ## NATIVE ## 02 :hover => isOpened => :active => color: 'red' 1`] = `null`;

exports[`SWITCH ## NATIVE ## 03 atomizeRuleset(isOpened => atomizeRuleset(color: 'red), isClosed => atomizeRuleset(color: 'green')) 1`] = `
Array [
  Array [
    "IF $switch: isOpened",
    "color: red /*.*/",
  ],
  Array [
    "IF $switch: isClosed",
    "color: green /*.*/",
  ],
]
`;

exports[`SWITCH ## NATIVE ## 04 variant's order: black-red-green-yellow-blue-brown 1`] = `
Array [
  Array [
    "color: black /*.*/",
  ],
  Array [
    "IF $switch: isOpened",
    "color: red /*./$switch.isOpened*/",
  ],
  Array [
    "IF $switch: isOpened",
    "IF $switch: isClosed",
    "color: green /*./$switch.isOpened/$switch.isClosed*/",
  ],
]
`;

exports[`SWITCH ## NATIVE ## 05 variant's order with array: $web => brown-blue-yellow-red-black 1`] = `
Array [
  Array [
    "IF $switch: isOpened",
    "color: red /*.[1]/$switch.isOpened[1]*/",
  ],
  Array [
    "color: black /*.[2]*/",
  ],
]
`;

exports[`SWITCH ## NATIVE ## 06 nesting PSEUDO and $switch 1`] = `null`;

exports[`SWITCH ## WEB ## 01 isOpened: color: 'red', isClosed: color: 'blue' 1`] = `
Array [
  Array [
    "IF $switch: isOpened",
    ".a { color:red /*./$switch.isOpened*/ }",
  ],
  Array [
    "IF $switch: isClosed",
    ".b { color:blue /*./$switch.isClosed*/ }",
  ],
]
`;

exports[`SWITCH ## WEB ## 02 :hover => isOpened => :active => color: 'red' 1`] = `
Array [
  Array [
    "IF $switch: isOpened",
    ".a:hover:active { color:red /*./$web/:hover/$switch.isOpened/$web*/ }",
  ],
]
`;

exports[`SWITCH ## WEB ## 03 atomizeRuleset(isOpened => atomizeRuleset(color: 'red), isClosed => atomizeRuleset(color: 'green')) 1`] = `
Array [
  Array [
    "IF $switch: isOpened",
    ".a { color:red /*.*/ }",
  ],
  Array [
    "IF $switch: isClosed",
    ".b { color:green /*.*/ }",
  ],
]
`;

exports[`SWITCH ## WEB ## 04 variant's order: black-red-green-yellow-blue-brown 1`] = `
Array [
  Array [
    ".d { color:black /*.*/ }",
  ],
  Array [
    "IF $switch: isOpened",
    ".b { color:red /*./$switch.isOpened*/ }",
  ],
  Array [
    "IF $switch: isOpened",
    "IF $switch: isClosed",
    ".a { color:green /*./$switch.isOpened/$switch.isClosed*/ }",
  ],
  Array [
    "IF $switch: isOpened",
    ".c { color:yellow /*./$switch.isOpened/$web*/ }",
  ],
  Array [
    ".f { color:blue /*./$web*/ }",
  ],
  Array [
    "IF $switch: isOpened",
    ".e { color:brown /*./$web/$switch.isOpened*/ }",
  ],
]
`;

exports[`SWITCH ## WEB ## 05 variant's order with array: $web => brown-blue-yellow-red-black 1`] = `
Array [
  Array [
    "IF $switch: isOpened",
    ".a { color:brown /*.[0]/$web[0]/$switch.isOpened*/ }",
  ],
  Array [
    ".b { color:blue /*.[0]/$web[1]*/ }",
  ],
  Array [
    "IF $switch: isOpened",
    ".c { color:yellow /*.[1]/$switch.isOpened[0]/$web*/ }",
  ],
  Array [
    "IF $switch: isOpened",
    ".d { color:red /*.[1]/$switch.isOpened[1]*/ }",
  ],
  Array [
    ".e { color:black /*.[2]*/ }",
  ],
]
`;

exports[`SWITCH ## WEB ## 06 nesting PSEUDO and $switch 1`] = `
Array [
  Array [
    "IF $switch: isOpened",
    ".a:hover:active { margin:10px /*./$web/:hover/$switch.isOpened/$web*/ }",
  ],
]
`;
